pkg/protocol/commitment.go

```diff
@@ -309,7 +301,7 @@ func (c *Commitment) deriveChain(parent *Commitment) func() {
 
 		// if we are not the main child of our parent, we spawn a new chain
 		if c != mainChild {
-			if currentChain == nil {
```

Comment:
Can we add some more comments on this function? This is critical to decide how and when we spawn new chains if new commitments are received and when we evict after advancing the root commitment.